modules/@apostrophecms/admin-bar/ui/apos/components/TheAposAdminBar.vue

```diff
@@ -677,22 +684,24 @@ export default {
     },
     async onRevertPublishedToPrevious(data) {
       try {
-        const doc = await apos.http.post(`${data.action}/${data._id}/revert-published-to-previous`, {
+        const response = await apos.http.post(`${data.action}/${data._id}/revert-published-to-previous`, {
           body: {},
           busy: true
         });
-        apos.notify('Restored previously published version.', {
+        apos.notify(response.message, {
           type: 'success',
           dismiss: true
         });
         // This handler covers all "undo publish" buttons, so make sure it's
         // for the context document before altering any admin bar state
         // because of it
-        if (doc.aposDocId === (this.context && this.context.aposDocId)) {
+        const aposDocId = data._id.replace(/:.*$/, '');
+        if (data._id.replace(/:.*$/, '') === (this.moduleOptions.context && this.moduleOptions.context._id.replace(/:.*$/, ''))) {
```

Comment:
You have `data._id.replace(/:.*$/, '')` set to `aposDocId` now, to replace this.

Reply:
It's odd that `context` is on `this.moduleOptions` since it isn't, afaik, an option on the module. It's the context in which this particular instance of the component has been loaded. So `this.context` seems better. But this component already uses `this.context`, though I can't figure out where that's being set. Maybe it isn't and could be used instead?

Generally speaking it doesn't seem like `moduleOptions` is a thing that should ever be touched once set.

Reply:
Good catch re: the extra replace call, I removed one.

Re: `context` being on `moduleOptions`, that's because the server gives it to us as module options, we wouldn't have any other way of knowing about it.

`this.context` does not appear in the module at this point, I think you started reviewing before I caught a rogue instance of that.

Reply:
Ah, I see. I didn't see `context` on the module options somehow. It still seems like that should be referenced, but not mutated. It seems better to mutate that copy that the component owns.

`this.context` is in several places in this component, starting with L272. It's set in `onDiscardDraft`, but I don't see where it's originally set up.

Reply:
Nevermind that last bit. I wrote that before I digested your other comments on the other line.